Detectors/TPC/qc/src/Tracks.cxx

```diff
@@ -130,23 +138,62 @@ bool Tracks::processTrack(const o2::tpc::TrackTPC& track)
     auto propagator = o2::base::Propagator::Instance(true);
     const int type = (track.getQ2Pt() < 0) + 2 * track.hasCSideClustersOnly();
     auto dcaHist = mMapHist[fmt::format("hDCAr_{}", types[type]).data()].get();
+    auto dcaHist_pT_pos = mMapHist["hDCArVsPtPos"].get();
+    auto dcaHist_eta_pos = mMapHist["hDCArVsEtaPos"].get();
+    auto dcaHist_nCluster_pos = mMapHist["hDCArVsNClsPos"].get();
+    auto dcaHist_pT_neg = mMapHist["hDCArVsPtNeg"].get();
+    auto dcaHist_eta_neg = mMapHist["hDCArVsEtaNeg"].get();
+    auto dcaHist_nCluster_neg = mMapHist["hDCArVsNClsNeg"].get();
 
-    if (propagator->getMatLUT() && propagator->hasMagFieldSet()) {
-      // ---| fill DCA histos |---
-      o2::gpu::gpustd::array<float, 2> dca;
-      const o2::math_utils::Point3D<float> refPoint{0, 0, 0};
-      o2::track::TrackPar propTrack(track);
-      if (propagator->propagateToDCABxByBz(refPoint, propTrack, 2.f, o2::base::Propagator::MatCorrType::USEMatCorrLUT, &dca)) {
-        const auto phi = o2::math_utils::to02PiGen(track.getPhi());
-        dcaHist->Fill(phi, dca[0]);
-      }
-    } else {
-      static bool reported = false;
-      if (!reported) {
-        LOGP(error, "o2::base::Propagator not properly initialized, MatLUT ({}) and / or Field ({}) missing, will not fill DCA histograms", (void*)propagator->getMatLUT(), (void*)propagator->hasMagFieldSet());
-        reported = true;
+    // set-up sampling for the DCA calculation
+    Double_t sampleProb = 1;
+
+    if (mSamplingFractionDCAr > 0) { // for now no SEED is given.
+      TRandom3 randomGenerator;
+      sampleProb = randomGenerator.Uniform(1);
+    }
+
+    if (sampleProb > (Double_t)(1. - mSamplingFractionDCAr)) {
+
+      if (propagator->getMatLUT() && propagator->hasMagFieldSet()) {
+        // ---| fill DCA histos |---
+        o2::gpu::gpustd::array<float, 2> dca;
+        const o2::math_utils::Point3D<float> refPoint{0, 0, 0};
+        o2::track::TrackPar propTrack(track);
+        if (propagator->propagateToDCABxByBz(refPoint, propTrack, 2.f, o2::base::Propagator::MatCorrType::USEMatCorrLUT, &dca)) {
+          const auto phi = o2::math_utils::to02PiGen(track.getPhi());
+          dcaHist->Fill(phi, dca[0]);
+          if (sign < 0.) {
+            dcaHist_pT_neg->Fill(pt, dca[0]);
+            if (pt < mCutMaxPtDCAr) {
+              dcaHist_eta_neg->Fill(eta, dca[0]);
+              dcaHist_nCluster_neg->Fill(nCls, dca[0]);
+            }
+          } else {
+            dcaHist_pT_pos->Fill(pt, dca[0]);
+            if (pt < mCutMaxPtDCAr) {
+              dcaHist_eta_pos->Fill(eta, dca[0]);
+              dcaHist_nCluster_pos->Fill(nCls, dca[0]);
+            }
+          }
```

Comment:
Together with the above changes, and use the pT cut also for the default dca vs phi plot.
I also think the the pt cut should be removing the smaller pts.
```suggestion
          dcaHist_pT_neg->Fill(pt, dca[0]);
          if (pt > mCutMinPtDCAr) {
            dcaHist->Fill(phi, dca[0]);
            dcaHist_eta->Fill(eta, dca[0]);
            dcaHist_nCluster->Fill(nCls, dca[0]);
          }
```